packages/ia-components/sandbox/tiles/TileComponent.jsx

```diff
@@ -45,9 +45,9 @@ export default class TileComponent extends React.Component {
       const item = this.props.item;
       const query = props.query || (item && item.query) || (member && member.query);
       const isCollection = (member.mediatype === 'collection');
-      const collection0 = member.collection0() || (item && item.metadata && item.metadata.collection[0]);
+      const collection0 = member.collection0() || (item && item.metadata && item.metadata.collection && item.metadata.collection[0]);
       const by = member.creator || member.creatorSorter || (item && item.metadata.creator); // Should be array
-      const collection = member.collection || (item && item.metadata.collection) || []; // Should be array (note fav-xxx dont have collection)
+      const collection = member.collection || (item && item.metadata && item.metadata.collection) || []; // Should be array (note fav-xxx dont have collection)
```

Comment:
let's leverage some destructuring..

Reply:
How could that line be written more clearly with deconstruction, its a complex expression assigned to a constant ? Any deconstruction is going to require superflous temporary constants, still require all thsoe tests and decrease readability ... feel free to propose an alternative.

Reply:
ok, i've really *really* disliked deconstruction since the start.

i've never seen another language doing this, and my brain would *full brakes*, pause, try to even remember how these stupid things work, before continuing code scans...

however!  over time, i have _just_ come around the corner to liking them -- and i think preferring them.

when there are 2+ elements, it _really_ becomes a big win (IMHO).  and now i've found myself wishing it was in other langs i use a lot.

eg:
`const { title, description } = metadata`

i find myself no longer wanting to write `description` twice -- and is starting to feel more like clutter, compared to deconstruction.  

$0.02

Reply:
Yes - I'm not pro, or anti, deconstruction - I use them myself especially for multiple items like your example, and for the results of a method/function.
<rant> What I really really hate, is some wet-behind-the-ears coder at AirBnB defining a set of eslint rules, that arbitrarily disallow some particular and perfectly valid / not-confusing coding pattern in favor of something new and shiny, and that then we are adopting those rules as if they are in some way better - I've come to appreciate elint for catching some errors but it needs pruning of entirely arbitrary preferences of whoever wrote them. </rant>

Reply:
That's the whole point of a style guide, though. It gives all of the developers a common style to develop on. It gets rid of a lot of these bike-shedding conversations during merge requests because the developers have agreed upon a standard.

A conversation can be had about whether or not to enable different features of it like destructuring, but we should follow the style guide that we've chosen.

Reply:
But those two lines don't come close to the one I've got. it makes absolutely no sense to take the line:
```
const collection = member.collection || (item && item.metadata && item.metadata.collection) || [];
```
And replace it with 6 lines, plus a tonne of commenting to explain what doing. 
```
const { metadata = {} } = item;
const { title, collection } = metadata;
// Have to move it out the way, as we have two fields called "collection" 
const itemcollection = collection; 
const { collection } = member;
// Move this one out the way
const membercollection = collection; 
const collection = membercollection || itemcollection || [];
```
This is *nuts* if we've got style guidelines which enforce writing s***y code like this. I strongly recommend that if we are insisting on following all rules in the style guide, that this particular rule get removed.